### ISSUE TITLE
Webhook: v1aplha* and v1beta1 have missing fields when converting to v1beta2

### ISSUE THREAD

**Issue by VictorMari** [tmforum-oda/oda-canvas#118]
## Long story short

Components deployed in v1beta1 canvas are not backwards compatible with v1beta2 due to a bug in the webhook converter.

## Description

<summary>Management and security fields are not converted to managementFunction and SecurityFunction</summary>

Component file extract from the following v1beta1 component:
[component file](https://github.com/tmforum-oda/oda-canvas/files/12928172/r1-prodcatalog-fixedri.component.txt)

````
management: 
    - name: metrics
      apitype: prometheus
      implementation: r1-productcatalog-sm
      path: /r1-productcatalog/metrics
      port: 4000    
security:
  controllerRole: Admin
  securitySchemes:
    bearerAuth:
      type: http
      scheme: bearer
      bearerFormat: JWT
  partyrole:
    specification: https://tmf-open-api-table-documents.s3.eu-west-1.amazonaws.com/OpenApiTable/4.0.0/swagger/TMF669-PartyRole-v4.0.0.swagger.json
      implementation: r1-partyroleapi
      apitype: openapi
      path: /r1-productcatalog/tmf-api/partyRoleManagement/v4
      developerUI: /r1-productcatalog/tmf-api/partyRoleManagement/v4/docs
      port: 8080
````
<summary>After deployment the following errors can be seen in the controler logs:</summary>
See full logs here: 

[Controller logs](https://github.com/tmforum-oda/oda-canvas/files/12928129/ccl-logs.txt)

````
[2023-10-17 11:33:35,639] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|coreAPIs|coreAPIs] Patching API: productcatalogmanagement
[2023-10-17 11:33:35,799] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|managementAPIs|managementAPIs] Handler called: 
[2023-10-17 11:33:35,799] ComponentOperator    [ERROR   ] [r1-productcatalog|component/r1-productcatalog|managementAPIs|managementAPIs] Unhandled exception: 'managementFunction': Traceback (most recent call last):
  File "componentOperator/componentOperator.py", line 179, in managementAPIs
    newManagementAPIs = spec['managementFunction']['exposedAPIs']
  File "/usr/local/lib/python3.7/site-packages/kopf/_cogs/structs/dicts.py", line 297, in __getitem__
    return resolve(self._src, self._path + (item,))
  File "/usr/local/lib/python3.7/site-packages/kopf/_cogs/structs/dicts.py", line 121, in resolve
    result = result[key]
KeyError: 'managementFunction'
[ccl-logs.txt](https://github.com/tmforum-oda/oda-canvas/files/12928124/ccl-logs.txt)

[2023-10-17 11:33:35,950] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|securityAPIs|securityAPIs] Handler called: 
[2023-10-17 11:33:35,950] ComponentOperator    [ERROR   ] [r1-productcatalog|component/r1-productcatalog|securityAPIs|securityAPIs] Unhandled exception: 'securityFunction': Traceback (most recent call last):
  File "componentOperator/componentOperator.py", line 255, in securityAPIs
    newSecurityAPIs = spec['securityFunction']['exposedAPIs']
  File "/usr/local/lib/python3.7/site-packages/kopf/_cogs/structs/dicts.py", line 297, in __getitem__
    return resolve(self._src, self._path + (item,))
  File "/usr/local/lib/python3.7/site-packages/kopf/_cogs/structs/dicts.py", line 121, in resolve
    result = result[key]
KeyError: 'securityFunction'

[2023-10-17 11:33:36,086] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|publishedEvents|publishedEvents] Handler called: 
[2023-10-17 11:33:36,226] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|subscribedEvents|subscribedEvents] Handler called: 
[2023-10-17 11:33:36,366] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|summary|summary] Handler called: 
[2023-10-17 11:33:36,366] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|summary|summary] Creating summary - completed API count: 1/1
[2023-10-17 11:33:36,366] ComponentOperator    [INFO    ] [r1-productcatalog|component/r1-productcatalog|summary|summary] Creating summary - deployment status: In-Progress-SecCon
[2023-10-17 11:33:36,509] kopf.objects         [ERROR   ] [components/r1-productcatalog] Handler 'security_client_add/status.summary/status.deployment_status' failed temporarily: Could not get partyrole path from component. Will retry.
````

This is the component after conversion 
[v1beta2component.txt](https://github.com/tmforum-oda/oda-canvas/files/12928212/v1beta2component.txt)

The values for the securityFunction and managementFunction have not been copied to the new component version.
## Environment

* Kopf version: 
* Kubernetes version: 
* Python version: 
* OS/platform: 

<details><summary>Python packages installed</summary>

```
```

**Comment by LesterThomas** [comment]
@VictorMari - I believe this is fixed in PR #133 
Can you re-test to ensure you are happy and then close this issue?

**Comment by LesterThomas** [comment]
Fixed with PR #133
